### PR TITLE
hal: nxp: Changes for LPC55S36 entropy

### PR DESCRIPTION
**Pull request by decsny**
HAL changes to make entropy from LPC55S36 Crypto Subsystem work in zephyr

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

### CODE REVIEW

File: mcux/mcux-sdk/components/css_pkc/src/platforms/LPC55S3x/inc/ip_platform.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #ifndef IP_PLATFORM_H
 #define IP_PLATFORM_H
 
-#include "board.h"
```
Comment (mmahadevan108) [inline mcux/mcux-sdk/components/css_pkc/src/platforms/LPC55S3x/inc/ip_platform.h:23]: Changes to the SDK should be sent to the SDK github repo ((https://github.com/NXPmicro/mcux-sdk) as a PR. This way we don't have to maintain local SDK changes inside Zephyr.

Comment (decsny) [reply mcux/mcux-sdk/components/css_pkc/src/platforms/LPC55S3x/inc/ip_platform.h:23]: PR submitted

Comment (mmahadevan108) [reply mcux/mcux-sdk/components/css_pkc/src/platforms/LPC55S3x/inc/ip_platform.h:23]: Please also update the PATCH section of the README file with details about the files changes so we don't lose these changes on an SDK Update

File: mcux/README [hunk 2]
```diff
@@ -58,3 +58,6 @@ License Link:
    https://spdx.org/licenses/BSD-3-Clause.html
 
 Patch List:
+  - Add cmake to include directories for lpc css
+  - change line 114 of mcuxClHash.c from NULL to 0
+  - change including board.h to fsl_common.h in LPC55S3x/inc/ip_platform.h
```
Comment (mmahadevan108) [inline mcux/README:63]: Please include the full path in the SDK so it is easy to find the file.

File: mcux/README [hunk 3]
```diff
@@ -58,3 +58,6 @@ License Link:
    https://spdx.org/licenses/BSD-3-Clause.html
 
 Patch List:
+  - Add cmake to include directories for lpc css
```
Comment (mmahadevan108) [inline mcux/README:61]: No need to list this.

**Review by dleach02 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**